jupyterhub/spawners.py

```diff
@@ -350,6 +350,14 @@ def get_pod_manifest(self):
             )
             self.init_containers.append(init_container)
 
+            self.lifecycle_hooks={
+                "preStop": {
+                    "exec": {
+                        "command": ["/bin/sh", "-c", "/usr/local/bin/pre-stop.sh"]
```

Comment:
This will fail on older images - could we do something like to make it backward-compatible?

`"command": ["/bin/sh", "-c", "/usr/local/bin/pre-stop.sh", "||", "true"]`?

Reply:
Good spot! I changed that and tested. All works nice.